src/main/webapp/script.js

```diff
@@ -1 +1,54 @@
-// This will be the JavaScript file in which I will create the list elements, post them onto page, etc.
\ No newline at end of file
+// This will be the JavaScript file in which I will create the list elements, post them onto page, etc.
+
+/** 
+ * Fetches google maps api key.
+ * 
+ * @returns None.
+ */
+async function getMapKey() {
+    const responseFromServer = await fetch('/google-map-key');
+    const apiKey = await responseFromServer.text();
+
+    placeMapRequest(apiKey)
+}
+
+
+/** 
+ * Dinamically creates and places needed objets (div, script) to host google map
+ * 
+ * @returns None.
+ */
+async function placeMapRequest(apiKey){
+    let div = document.createElement('div');
+    let scrpit = document.createElement('script');
+
+    div.id = 'map'
+    scrpit.type = 'text/javascript';
+    scrpit.src = 
+        'https://maps.googleapis.com/maps/api/js?key=' +
+        apiKey +
+        '&callback=initMap';
```

Comment:
Although you guys can change any function's name as needed, make sure this line is updated with the current name `initMap()` has.